libavformat/movenc.c

```diff
@@ -2472,8 +2472,8 @@ static int mov_write_stbl_tag(AVFormatContext *s, AVIOContext *pb, MOVMuxContext
     mov_write_stsc_tag(pb, track);
     mov_write_stsz_tag(pb, track);
     mov_write_stco_tag(pb, track);
-    if (track->cenc.aes_ctr && !(mov->flags & FF_MOV_FLAG_FRAGMENT)) {
-        ff_mov_cenc_write_stbl_atoms(&track->cenc, pb, 0);
+    if (track->cenc.encryption_scheme != MOV_ENC_NONE) {
```

Comment:
For cenc, we were checking against MOV_FLAG_FRAGMENT. Now that check is removed. What is the impact of that? Does it mean to write stbl_atom for fragmented mp4 too?

Also does stbl_atom is needed for both cenc and cbcs?

Reply:
`stbl` is always required in the `moov`, but this code is for writing atoms inside the `stbl`. For fmp4 these atoms are moved to the `traf` instead of the `stbl`.

I had to add the `FF_MOV_FLAG_FRAGMENT` check back (and in one other place), because the behavior is different using dashenc - it will write these atoms erroneously in the `moov` because it already has an auxiliary entry (there is a check not to write anything if no entries exist) at this point.